tuf/tuf.go

```diff
@@ -173,6 +173,15 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetAllRoles returns a list of all role entries for this TUF repo
+func (tr *Repo) GetAllRoles() ([]*data.Role, error) {
+	roles := tr.keysDB.GetAllRoles()
+	if len(roles) == 0 {
+		return nil, data.ErrNoRoles{}
```

Comment:
There might be no roles if nothing has been loaded yet - like if it hasn't been initialized, for instance.  I guess that isn't really an error - maybe it'd be clearer if this was named `GetAllLoadedRoles` and that could return an empty array?  As opposed to `GetAllRoles` which may imply all base roles which must be in a valid TUF repo?